identity/src/main/java/com/android/identity/credential/Credential.java

```diff
@@ -105,7 +106,7 @@ public class Credential {
 
     private NameSpacedData mNameSpacedData = new NameSpacedData.Builder().build();
     private KeystoreEngine mKeystoreEngine;
-    private LinkedHashMap<String, byte[]> mApplicationData = new LinkedHashMap<>();
+    private ApplicationData mApplicationData = new SimpleApplicationData();
```

Comment:
This variable should be of type `SimpleApplicationData` so you don't have to declare things like `serialize()` / `deserialize()` in the interface.